src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -229,6 +228,67 @@ def get_custom_templatetags_libraries(self) -> list[str]:
             f"{prefix}.objects_api.json_tags",
         ]
 
+    def update_payment_status(self, submission: "Submission", options: dict) -> None:
+        config = ObjectsAPIConfig.get_solo()
+        assert isinstance(config, ObjectsAPIConfig)
+        config.apply_defaults_to(options)
+
+        if not options["payment_status_update_json"]:
+            logger.warning(
+                "Skipping payment status update because not template was configured."
+            )
+            return
+
+        context = {
+            "variables": get_variables_for_context(submission),
+            "payment": {
+                "completed": submission.payment_user_has_paid,
+                "amount": submission.payments.sum_amount(),
+                "public_order_ids": ",".join(
+                    submission.payments.get_completed_public_order_ids()
+                ),
```

Comment:
This is repeated in the `register_submission` method, I don't know if it's worth moving it in a single method?

Reply:
probably just using a utility function at the module level `def get_payment_context_data(submission: Submission)` would be sufficient :)